packages/tupaia-web/src/features/Dashboard/DashboardMenu/SubscribeModal.tsx

```diff
@@ -3,185 +3,141 @@
-const ButtonGroup = styled.div`
-  padding-top: 2.5rem;
-  width: 100%;
-  display: flex;
-  justify-content: flex-end;
+  justify-content: space-between;
+  width: 42rem;
+  max-width: 100%;
+  min-height: 18rem;
+  padding: 3rem 1.5rem 0.5rem;
+  text-align: left;
 `;
 
-const SubscribeButton = styled(Button)`
-  margin-left: 1.2rem;
+const Button = styled(UIButton)`
+  text-transform: none;
+  font-size: 0.875rem;
+  padding: 0.3rem 1.1rem;
 `;
 
 const EmailInput = styled(TextField)<{ $disabled?: boolean }>`
-  width: 50%;
-  margin: auto;
-  @media screen and (max-width: ${MOBILE_BREAKPOINT}) {
-    width: 100%;
-  }
-
+  width: 21rem;
+  max-width: 100%;
+  margin: 0 auto 2rem;
   ${({ $disabled, theme }) =>
-    $disabled
-      ? `
-
-      .MuiInput-input {
-        color: ${theme.palette.text.secondary}; 
-      }
-        
-      .MuiInput-underline:before { border-bottom: 1px outset rgba(255, 255, 255, 0.7); transition: none;  }
-      .MuiInput-underline:hover:before { border-bottom: 1px outset rgba(255, 255, 255, 0.7); }
-      .MuiInput-underline:after { border-bottom-style: none; }  
-      `
-      : ''};
+    $disabled &&
+    `.MuiInput-input {
+    color: ${theme.palette.text.secondary};
+  }
+  .MuiInput-root:before,.MuiInput-root:hover:before, .MuiInput-root:after {
+    border-bottom: 1px solid ${theme.palette.divider};
+  }`}
 `;
 
 export const SubscribeModal = () => {
+  const [isSubmitting, setIsSubmitting] = useState(false);
   const { activeDashboard, subscribeModalOpen, toggleSubscribeModal } = useDashboard();
+  const queryClient = useQueryClient();
   const { entityCode, projectCode } = useParams();
-  const { refetch: refetchDashboards } = useDashboards(projectCode, entityCode);
   const { data: user, isLoggedIn, isLoading } = useUser();
   const mailingList = useDashboardMailingList();
   const isSubscribed = mailingList ? mailingList.isSubscribed : false;
   const defaultEmail = isLoggedIn ? user.email : '';
   const formContext = useForm({
     mode: 'onChange',
-    defaultValues: {
-      email: defaultEmail,
-    },
   });
-  const {
-    mutateAsync: subscribe,
-    error: subscribeError,
-    reset: resetSubscribe,
-  } = useSubscribeDashboard(projectCode, entityCode, activeDashboard?.code);
-  const {
-    mutateAsync: unsubscribe,
-    error: unsubscribeError,
-    reset: resetUnsubscribe,
-  } = useUnsubscribeDashboard(projectCode, entityCode, activeDashboard?.code);
+  const { mutateAsync: subscribe, error: subscribeError } = useSubscribeDashboard(
+    projectCode,
+    entityCode,
+    activeDashboard?.code,
+  );
+  const { mutateAsync: unsubscribe, error: unsubscribeError } = useUnsubscribeDashboard(
+    projectCode,
+    entityCode,
+    activeDashboard?.code,
+  );
 
   const handleSubmit = async data => {
+    setIsSubmitting(true);
     if (isSubscribed) {
       await unsubscribe(data);
     } else {
       await subscribe(data);
     }
-    refetchDashboards();
+    setIsSubmitting(false);
```

Comment:
Can this behaviour be handled by `isSubmitting` from the `useMutation`?

Reply:
Unfortunately, since we are using mutateAsync, the loading state is not provided. I thought about refactoring to use non-async mutate, but in the end I decided not to do it that way since it would be too much refactoring and I've already gone a bit beyond the scope of the ticket.

Reply:
Ah, my mistake, thanks for clarifying 🙏